activesupport/lib/active_support/deprecation/proxy_wrappers.rb

```diff
@@ -144,8 +159,15 @@ def target
           ActiveSupport::Inflector.constantize(@new_const.to_s)
         end
 
-        def warn(callstack, called, args)
-          @deprecator.warn(@message, callstack)
+        def const_missing(name)
+          if target.const_defined?(name)
+            ActiveSupport::Deprecation::DeprecatedConstantProxy.new("#{@old_const}::#{name}", "#{@new_const}::#{name}")
+          end
```

Comment:
I don't feel strongly about it, but I was imagining this would do the same `@deprecator.warn` as `method_missing` (and then a `const_get`) -- so the message would still be talking about the renamed constant, rather than the child.

The only reason I think that might be important is if the child has itself been renamed or otherwise deprecated... perhaps better to complain that you meant `Deprecatee::New` over `Deprecatee::Old`, plus the second complaint that you accessed `Deprecatee::New::Descendant` when you should be using `Deprecatee::New::BetterDescendant`, instead of building a message that explicitly claims `Deprecatee::New::Descendant` is a new good thing. 

---

On a separate minor note, I think this should have an `else super`?